src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1525,6 +1527,25 @@ private function dumpValue($value, $interpolate = true)
             }
 
             return sprintf('array(%s)', implode(', ', $code));
+        } elseif ($value instanceof LocatorArgument) {
+            $services = $value->getValues();
+            $servicesMap = array();
+            $getterCode = 'function ($id) {'."\n";
+            foreach ($services as $k => $ref) {
+                $servicesMap[] = sprintf("'%s' => '%s'", $k, $ref);
+                $singleGetter = sprintf('            if (\'%s\' === $id || \'%s\' === $id) {', $k, $ref);
+                $singleGetter .= sprintf("\n               return %s;", $this->getServiceCall((string) $ref, $ref));
+                $singleGetter .= "\n            }\n";
+                $getterCode .= $singleGetter;
+            }
+            $getterCode .= '        }';
+
+            $hasserCode = 'function ($id) {'."\n";
+            $hasserCode .= sprintf('            $references = array(%s);', implode(', ', $servicesMap));
```

Comment:
@Taluu I tried it and in fact we only need a string map here, while `var_export` dumps Reference objects. Also looping over references is actually needed for generating the getter code so I think it can be kept as is, unless another existing helper exists for such